sdks/python/apache_beam/io/avroio.py

```diff
@@ -242,4 +257,117 @@ def read_records(self, file_name, range_tracker):
         block = _AvroUtils.read_block_from_file(f, codec, schema_string,
                                                 sync_marker)
         for record in block.records():
-          yield record
+          if self._coder:
+            yield self._coder.decode(record)
+          else:
+            yield record
+
+  def default_output_coder(self):
+    return self._coder or super(_AvroSource, self).default_output_coder()
+
+
+_avro_codecs = {
+    fileio.CompressionTypes.UNCOMPRESSED: 'null',
+    fileio.CompressionTypes.ZLIB: 'deflate',
+    # fileio.CompressionTypes.SNAPPY: 'snappy',
+}
+
+
+class WriteToAvro(beam.transforms.PTransform):
+  """A ``PTransform`` for writing avro files."""
+
+  def __init__(self,
+               file_path_prefix,
+               schema=None,
```

Comment:
I think, instead of accepting a schema or coder here, we should accept a coder which must be an AvroCoder (we should introduce a new type AvroCoder with a property schema and provide a default implementation).